src/accelerate/utils/dataclasses.py

```diff
@@ -224,6 +224,20 @@ def list(cls):
         return list(map(str, cls))
 
 
+class DeprecatedFieldDescriptor:
+    def __init__(self, field_name, replaced_with):
+        self.field_name = field_name
+        self.replaced_with = replaced_with
+
+    def __get__(self, instance, owner):
+        warnings.warn(
+            f"The `{self.field_name}` of `{owner}` is deprecated and will be removed in v0.27.0. "
```

Comment:
As this has been in here for a long, long time, let's deprecate it in v1.0.0 please.